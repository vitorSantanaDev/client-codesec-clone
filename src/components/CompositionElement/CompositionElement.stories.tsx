import { ComponentStory, ComponentMeta } from '@storybook/react'
import CompositionElement from '.'

export default {
  title: 'CompositionElement',
  component: CompositionElement,
  args: {}
} as ComponentMeta<typeof CompositionElement>

export const Default: ComponentStory<typeof CompositionElement> = () => <CompositionElement/>
