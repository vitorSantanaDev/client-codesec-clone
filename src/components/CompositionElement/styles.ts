import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${() => css`
    width: 352px;
    height: 4px;

    background: #edeff0;
    box-shadow: 0px 0px 177px 4px rgba(232, 55, 82, 0.7),
      0px 0px 96px 18px rgba(232, 55, 82, 0.5),
      0px 0px 14px 1px rgba(232, 55, 82, 0.8), 0px 0px 4px 1px #e83752;
    border-radius: 8px;
  `}
`
