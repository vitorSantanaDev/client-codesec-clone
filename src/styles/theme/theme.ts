export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
