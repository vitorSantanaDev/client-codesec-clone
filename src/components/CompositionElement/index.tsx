import * as S from './styles'

const CompositionElement = () => <S.Wrapper />

export default CompositionElement
