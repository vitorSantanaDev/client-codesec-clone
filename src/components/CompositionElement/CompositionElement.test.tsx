import { renderTheme } from 'styles/theme/render-theme'
import CompositionElement from '.'

describe('<CompositionElement />', () => {
  it('should render the composition element', () => {
    const { container } = renderTheme(<CompositionElement />)
    expect(container.firstChild).toHaveStyle({
      width: '352px',
      height: '4px',
      background: '#edeff0',
      'border-radius': '8px'
    })
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
