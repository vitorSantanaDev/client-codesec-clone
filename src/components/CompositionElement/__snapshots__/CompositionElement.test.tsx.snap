// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompositionElement /> should render the composition element 1`] = `
.c0 {
  width: 352px;
  height: 4px;
  background: #edeff0;
  box-shadow: 0px 0px 177px 4px rgba(232,55,82,0.7), 0px 0px 96px 18px rgba(232,55,82,0.5), 0px 0px 14px 1px rgba(232,55,82,0.8),0px 0px 4px 1px #e83752;
  border-radius: 8px;
}

<div
  class="c0"
/>
`;
